### ISSUE TITLE
Add Arcane Engineering Into the Pack? (Tiny mod with a total of 9 items, technically 10 but whatever)

### ISSUE THREAD

**Issue by Alastors** [GTNewHorizons/GT-New-Horizons-Modpack#7873]
#### What do you suggest instead/what changes do you propose?

tl;dr: This is a very tiny addon with very big potential that would be incredibly useful in the long run to fork as it contains the potential for incredible modulability of additions to TC stuff if it were so wished! 

Okay so here is what I propose 

https://ftbwiki.org/Arcane_Engineering
https://www.curseforge.com/minecraft/mc-mods/arcane-engineering

Arcane Engineering is a *very* tiny TC addon made to bridge Thaumcraft and IE. No please don't leave, I'm not saying you should add IE into the pack. Instead I suggest a very tiny rewrite, by which I mean this addon uses very little materials from IE itself, and even with that, those that do get used are also materials that our modpack itself contains, that being Items such as steel, electrum, etc! 

You may be asking yourself, Alastor WTF does this addon add that would even justify doing that bs? It's quite simple, it adds one very large thing that I absolutely adore! Better wand modding! By that I mean it adds a wand body that, while having less capacity than a normal wooden body by itself, has the ability to store wand modifiers (items that change how certain aspects of the wand) inside an internal storage ui. Up until a point (being 4 to 6 slots depending on what cap you use, electrum adding 1, steel adding 2, normal caps adding no slots at all, however both the electrum and steel only have a meager vis discount of 5%)

That being said, the main feature of this mod is the modular wand. The wand itself would require a small shift in how you make it as of right now, as it uses the treated wood stick from IE as it's core base much as TC uses a stick as it's first wand's core. However, I genuinely believe this wouldn't be much of an issue to switch over! Perhaps we could use the sealed wood stick from base GT, or maybe we could instead use a magical wood tool rod or any other various whatever stick/rod shape you wish tbh it doesn't really matter because the modifier storage is so much more important than the wand itself!

What do these modifiers do you might ask? Here is a small list!
1. Vis Capacitors
           - These modifiers adds capacity to the wand, the first puts the wand capacity up to 45, the second puts the capacity up to 67, the third puts the capacity up to 101, and the fourth and final one (the wand itself can have more than four slots but it can only accept a maximum of 4 vis capacitors) brings capacity up to 151 being just slightly better than a silverwood wand.
           - My suggestion on this one is that perhaps instead of using the vis capacitors as is perhaps we could switch out the number of capacitors modifiers with batteries and have the wand continue to grow with the player such that say the lv battery brought it up to a wooden wand, the mv/hv battery brought it up to greatwood, the say ev battery brought it up to silverwood capacity, and so forth and so on. This change would only take up one slot but I think it would bring a lot of fun to the pack!

2. Focus Energizers
          - These modifiers add a level of potency onto foci up until a max total of 4 of these upgrades, making the foci equipped to the wand more powerful as suggested.
          - My suggestion for this one? Just leave it alone tbh, yes go and edit it's recipe but, at the core of it, it really isn't all that overpowered as Foci in there current form really arent all that good.
          - And thats anther point! Foci arent all that good right now! So this would give those wishing to use more magic in this pack a reason to use foci over their crossbow! Something that I think we can all agree would be a good thing as the progression has pretty much just become "yeah make a crossbow and just you know occasionally upgrade it, its really op right now and there isn't a reason to make any other weapon" so the capability for more variety in weapon choice would be a much beneficial addition!

3. Crystalline Charger
         - In my opinion this is the bread and butter of the reasoning you should want to add this to the pack, a lot of people will note that recharging your wand is pretty much just the same old same old, just put that shit on a pillar that gets infused with energized node energy or bring primal charms with you, or break some mushrooms. And I think we can all agree that is boring as all hell! Thats where the Crystalline Charger comes in! It's main feature is that it recharges the modular wand's internal vis buffer at increasing rates depending on how many crystalline chargers (up to the max of 4 chargers), very much mirroring the ability of the amber wand that is currently in the pack with the capacity to improve vis storage above the amber's measly 15 vis cap.
         - My suggestion for this part would be to replace it with a various tiered part from GT, such a way that I suggested for the Vis capacitors. That being, lets use circuits for this example, the lv circuit maybe charging at a twentieth of the recharge rate of the amber wand core and so forth and so on improving each tier until you create a wand that recharges much faster than the warpwood's warp based recharge rate toward the end of the game.

4. Something more?
          - I have one final suggestion for a potential rework to work with this pack. That being the potential for y'all to add in custom modifications! Now you may wonder "What do you mean custom modifications? You've been talking about that the entire time!?" No, what I mean is adding this mod would give y'all the potential to add in your own gregtech modifications to the wand itself to allow it to do more than it already can! To be honest the only reason I make this suggest is the fact that almost every suggestion I have made before reduces the usage of slots pretty severely as you could probably tell, which gave me this idea! Perhaps you could add more vis discounts by switching in different tiered wires, with some adding more discount and some taking away discount, or maybe instead switching this to lenses instead to give them more of a use? Maybe you could add mods that would give a buff to the player if they had the wand in their inventory? Maybe you could even add a modification that would give two or more wands the ability to tie themselves and share vis storage or something? My point being that the potential for adding new stuff to TC through this mod would be as unrestricted as the potential wand mod capacities of the wand itself, only really limiting y'all to whatever modifier potential y'all felt like adding into the wands of the game through this!

So here's my point, this is an incredibly tiny mod that would take very little time and work to bring into GTNH standards, and yet even with that, it adds so much variety and future possibilities for potential easier buff fixes to TC, that in my opinion it would be so much more valuable than the little time needed to create the fork! 

#
#### What is your GTNH Discord username?
Alastors_Game

**Comment by Prometheus0000** [comment]
Licence on CF says GNU, but who trusts that? 

> Sadly, I lost the source code of this mod when I did stupid things with my harddrive, and even more stupidly never made one single backup in my entire life, even after I lost all of my data because I never learn, do I.

There's no source code, so any changes mean decompiling.

**Comment by Dream-Master** [comment]
Required Dependency:

Thaumcraft
Immersive Engineering

Means we need to recode it first before we can use it.

**Comment by Dream-Master** [comment]
> Licence on CF says GNU, but who trusts that?
> 
> > Sadly, I lost the source code of this mod when I did stupid things with my harddrive, and even more stupidly never made one single backup in my entire life, even after I lost all of my data because I never learn, do I.
> 
> There's no source code, so any changes mean decompiling.

If it is GNU License it is open.

**Comment by Prometheus0000** [comment]
> Means we need to recode it first before we can use it.

Elisis said they'd make a decompiled fork for us.

> If it is GNU License it is open.

I've been told not to trust curseforge about licenses from mitch before.

**Comment by Dream-Master** [comment]
I will ask the mod dev to hand out us the code . Maybe that worked too.

**Comment by Alastors** [comment]
We have the code, I'll see if I actually have time to convert the crap soon

**Comment by Dream-Master** [comment]
I am not willing to add imersive engery so not sure what are left from a crossover mod.

**Comment by Alastors** [comment]
No you wouldn't be stupid, you're just going to make a better mod, god just grow psionic future sight already 🤦
